@now/go can't build app with private git

I'm using Go Modules. If my repo is a public it's run perfectly. But when I changed it to private it throws an error.

Here's the logs:
```bash
go get: warning: modules disabled by GO111MODULE=auto in GOPATH/src;
    ignoring ../../go.mod;
    see 'go help modules'
# cd .; git clone https://github.com/rohmanhm/aturmakan /tmp/494232d1/src/github.com/rohmanhm/aturmakan
Cloning into '/tmp/494232d1/src/github.com/rohmanhm/aturmakan'...
fatal: could not read Username for 'https://github.com': terminal prompts disabled
package github.com/rohmanhm/aturmakan/i18n: exit status 128
```

I've asked on spectrum but no answer yet.
https://spectrum.chat/zeit/now/now-doesnt-work-with-private-git-and-go-modules~8daedb5f-f5a3-4ac1-a385-578265e14c51

@sophearak Can you help me out getting started? The `now.json` in the affected project looks like this:
```json
{
  "version": 2,
  "builds": [{ "src": "*.go", "use": "git+https://github.com/marhaupe/now-go"}]
}
````
As far as I understand, now will simply run `npm install git+https://github.com/marhaupe/now-go` and use this builder. This fails however. My logs look like this: 

```
@zeit/now-dcs-info:` using https://dcs.now.systems as the upstream
--
BUILD | May 15 2019 13:29:29:508 | running yarn info for @now/build-utils...
BUILD | May 15 2019 13:29:30:700 | running yarn for @now/build-utils...
BUILD | May 15 2019 13:29:30:939 | yarn add v1.13.0
BUILD | May 15 2019 13:29:30:959 | warning package.json: No license field
BUILD | May 15 2019 13:29:30:969 | info No lockfile found.
BUILD | May 15 2019 13:29:31:009 | warning No license field
BUILD | May 15 2019 13:29:31:011 | [1/4] Resolving packages...
BUILD | May 15 2019 13:29:31:488 | [2/4] Fetching packages...
BUILD | May 15 2019 13:29:32:342 | [3/4] Linking dependencies...
BUILD | May 15 2019 13:29:32:628 | [4/4] Building fresh packages...
BUILD | May 15 2019 13:29:32:640 | success Saved lockfile.
BUILD | May 15 2019 13:29:32:642 | warning No license field
BUILD | May 15 2019 13:29:32:652 | success Saved 45 new dependencies.
BUILD | May 15 2019 13:29:32:653 | info Direct dependencies
BUILD | May 15 2019 13:29:32:654 | └─ @now/build-utils@0.5.5
BUILD | May 15 2019 13:29:32:654 | info All dependencies
BUILD | May 15 2019 13:29:32:654 | ├─ @now/build-utils@0.5.5
BUILD | May 15 2019 13:29:32:654 | ├─ @types/cross-spawn@6.0.0 ├─ @types/node@12.0.2 ├─ async-retry@1.2.3 ├─ async-sema@2.1.4 ├─ balanced-match@1.0.0 ├─ brace-expansion@1.1.11
BUILD | May 15 2019 13:29:32:654 | ├─ buffer-crc32@0.2.13 ├─ concat-map@0.0.1 ├─ core-util-is@1.0.2 ├─ cross-spawn@6.0.5 ├─ double-ended-queue@2.1.0-0 ├─ end-of-stream@1.4.1 ├─ errno@0.1.7 ├─ from2@2.3.0 ├─ fs-extra@7.0.0
BUILD | May 15 2019 13:29:32:655 | ├─ fs.realpath@1.0.0 ├─ glob@7.1.3 ├─ graceful-fs@4.1.15 ├─ inflight@1.0.6 ├─ into-stream@5.0.0 ├─ isarray@1.0.0 ├─ isexe@2.0.0
BUILD | May 15 2019 13:29:32:655 | ├─ jsonfile@4.0.0 ├─ memory-fs@0.4.1 ├─ minimatch@3.0.4 ├─ multistream@2.1.1 ├─ nice-try@1.0.5 ├─ node-fetch@2.2.0 ├─ p-is-promise@2.1.0
BUILD | May 15 2019 13:29:32:655 | ├─ path-is-absolute@1.0.1 ├─ path-key@2.0.1 ├─ process-nextick-args@2.0.0 ├─ prr@1.0.1 ├─ readable-stream@2.3.6 ├─ retry@0.12.0 ├─ safe-buffer@5.1.2 ├─ semver@5.7.0 ├─ shebang-command@1.2.0
BUILD | May 15 2019 13:29:32:655 | ├─ shebang-regex@1.0.0 ├─ string_decoder@1.1.1 ├─ universalify@0.1.2 ├─ util-deprecate@1.0.2 ├─ which@1.3.1 └─ yazl@2.4.3
BUILD | May 15 2019 13:29:32:657 | Done in 1.73s.
BUILD | May 15 2019 13:29:32:669 | running yarn for git+https://github.com/marhaupe/now-go...
BUILD | May 15 2019 13:29:32:925 | yarn add v1.13.0
BUILD | May 15 2019 13:29:32:945 | warning package.json: No license field
BUILD | May 15 2019 13:29:32:956 | info No lockfile found.
BUILD | May 15 2019 13:29:32:995 | warning No license field
BUILD | May 15 2019 13:29:32:997 | [1/4] Resolving packages...
BUILD | May 15 2019 13:29:34:068 | [2/4] Fetching packages...
BUILD | May 15 2019 13:29:35:172 | [3/4] Linking dependencies...
BUILD | May 15 2019 13:29:35:377 | [4/4] Building fresh packages...
BUILD | May 15 2019 13:29:35:388 | success Saved lockfile.
BUILD | May 15 2019 13:29:35:390 | warning No license field
BUILD | May 15 2019 13:29:35:400 | success Saved 34 new dependencies.
BUILD | May 15 2019 13:29:35:400 | info Direct dependencies
BUILD | May 15 2019 13:29:35:401 | └─ now-go@0.4.6
BUILD | May 15 2019 13:29:35:401 | info All dependencies
BUILD | May 15 2019 13:29:35:401 | ├─ chownr@1.1.1
BUILD | May 15 2019 13:29:35:402 | ├─ cross-spawn@6.0.5 ├─ debug@4.1.1 ├─ end-of-stream@1.4.1 ├─ execa@1.0.0 ├─ fs-extra@7.0.1 ├─ fs-minipass@1.2.5 ├─ get-stream@4.1.0 ├─ graceful-fs@4.1.15
BUILD | May 15 2019 13:29:35:402 | ├─ is-stream@1.1.0 ├─ isexe@2.0.0 ├─ jsonfile@4.0.0 ├─ minimist@0.0.8 ├─ minizlib@1.2.1 ├─ mkdirp@0.5.1 ├─ ms@2.1.1 ├─ nice-try@1.0.5 ├─ node-fetch@2.5.0
BUILD | May 15 2019 13:29:35:402 | ├─ now-go@0.4.6 ├─ npm-run-path@2.0.2 ├─ once@1.4.0 ├─ p-finally@1.0.0 ├─ path-key@2.0.1 ├─ pump@3.0.0
BUILD | May 15 2019 13:29:35:402 | ├─ semver@5.7.0 ├─ shebang-command@1.2.0 ├─ shebang-regex@1.0.0 ├─ signal-exit@3.0.2 ├─ strip-eof@1.0.0 ├─ tar@4.4.6 ├─ universalify@0.1.2 ├─ which@1.3.1
BUILD | May 15 2019 13:29:35:402 | ├─ wrappy@1.0.2 └─ yallist@3.0.3
BUILD | May 15 2019 13:29:35:404 | Done in 2.49s.
BUILD | May 15 2019 13:29:35:429 | { Error: Cannot find module '/tmp/utils/build-module/node_modules/now-go'     at Function.Module._resolveFilename (module.js:547:15)     at Function.Module._load (module.js:474:25)     at Module.require (module.js:596:17)     at require (internal/module.js:11:18)     at installationStep (/var/task/worker.js:26586:16)     at <anonymous> code: 'MODULE_NOT_FOUND' }```

I'm clueless to what's going on here. Can you point me in the right direction? Am I doing something obviously wrong?

--- a/packages/now-go/index.ts
+++ b/packages/now-go/index.ts
@@ -1,5 +1,6 @@
 import { join, sep, dirname, basename } from 'path';
 import { readFile, writeFile, pathExists, move, copy } from 'fs-extra';
+import { homedir } from 'os';
 import execa from 'execa';
 
 import {
@@ -36,7 +37,7 @@ async function initPrivateGit(username: string, token: string, host: string) {
     'config',
     '--global',
     'credential.helper',
-    `store --file ${path.join(os.homedir(), '.git-credentials')`,
+    `store --file ${join(homedir(), '.git-credentials')}`,
   ]);
 
   await writeFile(

--- a/packages/now-go/index.ts
+++ b/packages/now-go/index.ts
@@ -1,5 +1,6 @@
 import { join, sep, dirname } from 'path';
 import { readFile, writeFile, pathExists, move } from 'fs-extra';
+import execa from 'execa';
 
 import {
   glob,
@@ -29,6 +30,21 @@ interface BuildParamsType extends BuildOptions {
   meta: BuildParamsMeta;
 }
 
+// Initialize private git repo for Go Modules
+async function initPrivateGit(username: string, token: string, host: string) {
+  await execa('git', [
+    'config',
+    '--global',
+    'credential.helper',
+    `store --file ${process.env.HOME}/.git-credentials`,
+  ]);
+
+  await writeFile(
+    `${process.env.HOME}/.git-credentials`,
+    `https://${username}:${token}@${host}`
+  );
+}
+
 export const version = 2;
 
 export const config = {
@@ -41,6 +57,19 @@ export async function build({
   config,
   meta = {} as BuildParamsMeta,
 }: BuildParamsType) {
+  if (
+    process.env.GIT_USERNAME &&
+    process.env.GIT_TOKEN &&
+    process.env.GIT_HOST
+  ) {
+    console.log('Initialize Git...');
+    await initPrivateGit(
+      process.env.GIT_USERNAME,
+      process.env.GIT_TOKEN,
+      process.env.GIT_HOST
+    );
+  }
+
   console.log('Downloading user files...');
   const entrypointArr = entrypoint.split(sep);
 

--- a/packages/now-go/index.ts
+++ b/packages/now-go/index.ts
@@ -32,7 +32,7 @@ interface BuildParamsType extends BuildOptions {
 }
 
 // Initialize private git repo for Go Modules
-async function initPrivateGit(username: string, token: string, host: string) {
+async function initPrivateGit(credentials: string) {
   await execa('git', [
     'config',
     '--global',
@@ -40,10 +40,7 @@ async function initPrivateGit(username: string, token: string, host: string) {
     `store --file ${join(homedir(), '.git-credentials')}`,
   ]);
 
-  await writeFile(
-    `${process.env.HOME}/.git-credentials`,
-    `https://${username}:${token}@${host}`
-  );
+  await writeFile(join(homedir(), '.git-credentials'), credentials);
 }
 
 export const version = 2;
@@ -59,17 +56,9 @@ export async function build({
   workPath,
   meta = {} as BuildParamsMeta,
 }: BuildParamsType) {
-  if (
-    process.env.GIT_USERNAME &&
-    process.env.GIT_TOKEN &&
-    process.env.GIT_HOST
-  ) {
-    console.log('Initialize Git...');
-    await initPrivateGit(
-      process.env.GIT_USERNAME,
-      process.env.GIT_TOKEN,
-      process.env.GIT_HOST
-    );
+  if (process.env.GIT_CREDENTIALS) {
+    console.log('Initialize Git credentials...');
+    await initPrivateGit(process.env.GIT_CREDENTIALS);
   }
 
   console.log('Downloading user files...');

--- a/packages/now-go/index.ts
+++ b/packages/now-go/index.ts
@@ -36,7 +36,7 @@ async function initPrivateGit(username: string, token: string, host: string) {
     'config',
     '--global',
     'credential.helper',
-    `store --file ${process.env.HOME}/.git-credentials`,
+    `store --file ${path.join(os.homedir(), '.git-credentials')`,
   ]);
 
   await writeFile(

--- a/packages/now-go/index.ts
+++ b/packages/now-go/index.ts
@@ -56,7 +56,7 @@ export async function build({
   workPath,
   meta = {} as BuildParamsMeta,
 }: BuildParamsType) {
-  if (process.env.GIT_CREDENTIALS) {
+  if (process.env.GIT_CREDENTIALS && !meta.isDev) {
     console.log('Initialize Git credentials...');
     await initPrivateGit(process.env.GIT_CREDENTIALS);
   }

